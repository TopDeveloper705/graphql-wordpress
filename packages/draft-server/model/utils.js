import slugifyUtil from 'slugify';

// eslint-disable-next-line
const removePattern = /[#,$*_+~.()\[\]\/'"!\-:@]/g;

export function slugify(value) {
  return slugifyUtil(value, {
    lower: true,
    remove: removePattern,
  });
}

// eslint-disable-next-line
export async function getUniqueSlug(collection, slugToCheck) {
  const slugified = slugify(slugToCheck);
  let i = 0;
  let slug = slugified;

  const count = async checkSlug => collection.count({ slug: checkSlug });
  // eslint-disable-next-line no-await-in-loop
  while ((await count(slug)) > 0) {
    i += 1;
    slug = `${slugified}-${i}`;
  }
  return slug;
}
