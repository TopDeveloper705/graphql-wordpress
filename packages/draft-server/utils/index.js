import fs from 'fs';
import path from 'path';

export function requireModules(dir) {
  const modules = {};
  const fileDir = path.resolve(dir);
  const files = fs.readdirSync(fileDir);
  files.forEach(file => {
    if (file.charAt(0) !== file.charAt(0).toUpperCase()) {
      return;
    }
    const [name] = file.split('.');
    // eslint-disable-next-line import/no-dynamic-require, global-require
    const module = require(path.join(fileDir, name));
    modules[name] = module.default;
  });
  return modules;
}

export function requireFiles(dir) {
  const files = {};
  const fileDir = path.resolve(dir);
  fs.readdirSync(fileDir).forEach(file => {
    if (file.charAt(0) !== file.charAt(0).toUpperCase()) {
      return;
    }
    const [name] = file.split('.');
    const filename = path.join(fileDir, file);
    files[name] = fs.readFileSync(filename, 'utf8');
  });
  return files;
}
