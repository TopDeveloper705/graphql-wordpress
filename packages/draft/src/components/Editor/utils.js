// eslint-disable-next-line
export function findWithRegex(regex, contentBlock, callback) {
  const text = contentBlock.getText();
  let matchArr;
  let start;
  // eslint-disable-next-line
  while ((matchArr = regex.exec(text)) !== null) {
    start = matchArr.index;
    callback(start, start + matchArr[0].length);
  }
}
