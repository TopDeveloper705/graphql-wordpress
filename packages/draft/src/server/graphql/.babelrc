{
  "presets": [
    ["kyt-core", {"modules": true}]
  ],
  "plugins": [
    "transform-class-properties",
    "transform-es2015-modules-commonjs",
    "transform-object-rest-spread",
    "transform-runtime",
    ["module-resolver", {
      "root": ["./src"]
    }]
  ]
}
