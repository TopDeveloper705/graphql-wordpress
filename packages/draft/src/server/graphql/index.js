import 'dotenv/config';
import express from 'express';
import { graphqlExpress, graphiqlExpress } from 'apollo-server-express';
import bodyParser from 'body-parser';
import { makeExecutableSchema } from 'graphql-tools';
import { MongoClient } from 'mongodb';
import cors from 'cors';
import typeDefs from 'server/graphql/schema';
import resolvers from 'server/graphql/resolvers';
import addModelsToContext from './models';
import authenticate from './authenticate';

/* eslint-disable no-console */

const schema = makeExecutableSchema({ typeDefs, resolvers });

const { GRAPHQL_PORT = 8080, MONGO_URL, MONGO_DB } = process.env;

async function startServer() {
  const client = await MongoClient.connect(MONGO_URL);
  const db = client.db(MONGO_DB);

  const app = express().use('*', cors());
  app.use(bodyParser.urlencoded({ extended: true }));
  app.use(bodyParser.json());

  app.use((req, res, next) => {
    req.context = addModelsToContext({ db });

    next();
  });

  authenticate(app);

  app.use(
    '/graphql',
    graphqlExpress(req => ({
      schema,
      context: req.context,
    }))
  );

  app.use(
    '/graphiql',
    graphiqlExpress({
      endpointURL: '/graphql',
    })
  );

  app.listen(GRAPHQL_PORT, () =>
    console.log(`API Server is now running on http://localhost:${GRAPHQL_PORT}`)
  );
}

startServer()
  .then(() => {
    console.log('All systems go');
  })
  .catch(e => {
    console.error('Uncaught error in startup');
    console.error(e);
    console.trace(e);
  });
