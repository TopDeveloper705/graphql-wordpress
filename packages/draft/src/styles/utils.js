import styled from 'react-emotion';
import { css } from 'emotion';
import facepaint from 'facepaint';
import themeUtils from 'styles/theme';
import responsive from 'styles/responsive';

export const headingStyles = css`
  display: block;
  font-weight: ${themeUtils.fonts.weight.bold};
  letter-spacing: 0.3px;
  line-height: 1.3;
  margin-bottom: 10px;

  ${responsive.desktop} {
    margin-bottom: ${themeUtils.padding}px;
  }
`;

export const h1styles = css`
  ${headingStyles};
  font-size: 30px;

  ${responsive.desktop} {
    font-size: 36px;
  }
`;

export const h2Styles = css`
  ${headingStyles};
  font-family: ${themeUtils.fonts.futura};
  font-size: 24px;
`;

export const h3Styles = css`
  ${headingStyles};
  font-family: ${themeUtils.fonts.futura};
  font-size: 20px;
`;

export const h4Styles = css`
  ${headingStyles};
  font-family: ${themeUtils.fonts.futura};
  font-size: 18px;
`;

export const Heading = styled.h2`
  ${headingStyles};
  font-family: ${themeUtils.fonts.futura};
  font-size: 25px;
`;

export const LoadMore = styled.button`
  appearance: none;
  background: ${themeUtils.colors.white};
  border: 1px solid ${themeUtils.colors.detail};
  box-sizing: border-box;
  color: ${themeUtils.colors.inactive};
  cursor: pointer;
  font-size: 16px;
  height: 32px;
  line-height: 1;
  text-align: center;
  text-transform: uppercase;
  transition: 400ms;
  width: 80px;

  &:hover,
  &:active,
  &:focus {
    border: 1px solid ${themeUtils.colors.black};
    color: ${themeUtils.colors.black};
    outline: 0 none;
  }
`;

export const buttonStyles = css`
  appearance: none;
  border-radius: 3px;
  box-sizing: border-box;
  cursor: pointer;
  display: inline-block;
  font-size: 13px;
  height: 28px;
  line-height: 26px;
  margin: 0;
  outline: 0;
  padding: 0 10px 1px;
  text-decoration: none;
  transition-duration: 0.05s;
  transition-property: border, background, color;
  transition-timing-function: ease-in-out;
  vertical-align: top;
  white-space: nowrap;

  &:active {
    transform: translateY(1px);
  }
`;

const pseudo = facepaint([':active', ':focus', ':hover']);

const generateButtonColors = c =>
  css(
    pseudo({
      background: [c.background, c.active.background, c.focus.background, c.hover.background],
      borderColor: [c.border, c.active.border, c.focus.border, c.hover.border],
      boxShadow: [c.boxShadow, c.active.boxShadow, c.focus.boxShadow],
      color: [c.color, null, c.focus.color, c.hover.color],
    })
  );

export const buttonColors = generateButtonColors(themeUtils.buttons.base);

export const Button = styled.button`
  ${buttonStyles};
  ${buttonColors};
`;

const largeButtonStyles = css`
  ${buttonStyles};
  height: 30px;
  line-height: 28px;
  padding: 0 12px 2px;
`;

const primaryButtonColors = generateButtonColors(themeUtils.buttons.primary);

export const PrimaryButton = styled.button`
  ${largeButtonStyles};
  ${primaryButtonColors};
  text-shadow: ${themeUtils.buttons.primary.textShadow};
`;

const smallButtonStyles = css`
  ${buttonStyles};
  font-size: 11px;
  height: 24px;
  line-height: 22px;
  padding: 0 8px 1px;
`;

export const SecondaryButton = styled.button`
  ${smallButtonStyles};
  ${buttonColors};
`;
