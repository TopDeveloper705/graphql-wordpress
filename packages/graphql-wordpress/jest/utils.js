import fs from 'fs';
import path from 'path';
import { makeExecutableSchema, addMockFunctionsToSchema } from 'graphql-tools';

const schemaFile = path.resolve('./generated/schema.graphql');

export const getMockSchema = (mocks = {}, preserveResolvers = false) => {
  const graphqlSchema = fs.readFileSync(schemaFile, 'utf8');
  const schema = makeExecutableSchema({ typeDefs: graphqlSchema });
  // Add mocks, modifies schema in place
  addMockFunctionsToSchema({ schema, mocks, preserveResolvers });
  return schema;
};

export const dateRegex = /[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}/;
