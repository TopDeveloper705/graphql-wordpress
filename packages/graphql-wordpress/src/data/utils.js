import url from 'url';
import request from 'request-promise';
import getClient from 'data/client';

/* eslint-disable no-console */

export const toBase64 = str => new Buffer(str).toString('base64');

const rp = (path, opts = {}) => {
  let uri = path;
  const urlObj = url.parse(path);
  if (!urlObj.host) {
    uri = `${process.env.WP_API_HOST}${path}`;
  }

  const params = Object.assign(
    {
      uri,
      jar: true,
      json: true,
      simple: true,
      strictSSL: false,
    },
    opts
  );

  return request(params);
};

export const clearEndpointCache = path => {
  const client = getClient();
  // eslint-disable-next-line no-console
  console.log(`Clearing cache for: ${path}`);
  return new Promise((resolve, reject) => {
    client.hkeys(path, (err, keys) => {
      if (err) {
        console.log(`Cache lookup failed for: ${path}`);
        return reject(err);
      }
      return resolve(keys.map(hashKey => client.hdelAsync(path, hashKey)));
    });
  });
};

const fetchData = (path, opts = {}) => {
  const key = toBase64(`${JSON.stringify(opts)}`);
  const client = getClient();

  const params = {
    ...opts,
    resolveWithFullResponse: true,
  };

  return new Promise((resolve, reject) => {
    const method = params.method || 'GET';
    const makeRequest = (mutation = false) =>
      rp(path, params)
        .catch(error => {
          console.log(`${method} Request failed for: ${path}`);
          console.log(error);
          reject(error);
        })
        .then(async response => {
          if (response) {
            console.log(`Made ${method} request for: ${path}`);
            if (mutation) {
              await clearEndpointCache(path);
            }
            client.hset(path, key, JSON.stringify(response));
            resolve({
              cache: 'miss',
              data: response,
            });
          } else {
            reject();
          }
        });

    if (params.method && params.method !== 'GET') {
      makeRequest(true);
    } else {
      client.hget(path, key, (err, cached) => {
        if (err) {
          reject(err);
        } else if (cached) {
          console.log(`Resolving from cache: ${path}`);
          resolve({
            cache: 'hit',
            data: JSON.parse(cached),
          });
        } else {
          makeRequest();
        }
      });
    }
  });
};

export default fetchData;
