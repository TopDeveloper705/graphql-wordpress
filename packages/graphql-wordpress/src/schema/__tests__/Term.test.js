import { graphql } from 'graphql';
import { getMockSchema } from 'jest/utils';
import getLoaders from 'data/loaders';

const context = {};
const rootValue = {
  loaders: getLoaders(),
};

const fields = `
  name
  count
  description
  slug
  taxonomy {
    rewrite {
      slug
    }
  }
`;

const queryTerm = `
  query TermQuery($slug: String!, $taxonomy: String!) {
    viewer {
      term(slug: $slug, taxonomy: $taxonomy) {
        ${fields}
      }
    }
  }
`;

describe('Test Term queries', () => {
  test('GraphQL should return a category by slug and taxonomy', async () => {
    const schema = getMockSchema({
      TermInterface: () => ({
        __typename: 'Category',
      }),
      Category: () => ({
        count: 200,
        name: 'Watch This',
        slug: 'watch-this',
        description: '',
      }),
      Rewrite: () => ({
        slug: 'category',
      }),
    });

    const variables = {
      slug: 'watch-this',
      taxonomy: 'category',
    };
    const result = await graphql(schema, queryTerm, rootValue, context, variables);
    expect(result).toMatchSnapshot();
  });

  test('GraphQL should return a tag by slug and taxonomy', async () => {
    const schema = getMockSchema({
      TermInterface: () => ({
        __typename: 'Tag',
      }),
      Tag: () => ({
        count: 3,
        name: '285 Kent',
        slug: '285-kent',
        description: '',
      }),
      Rewrite: () => ({
        slug: 'tag',
      }),
    });

    const variables = {
      slug: 'dirty-projectors',
      taxonomy: 'tag',
    };
    const result = await graphql(schema, queryTerm, rootValue, context, variables);
    expect(result).toMatchSnapshot();
  });
});
