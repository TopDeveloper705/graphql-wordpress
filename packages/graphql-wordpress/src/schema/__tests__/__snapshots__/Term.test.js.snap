// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Term queries GraphQL should return a category by slug and taxonomy 1`] = `
Object {
  "data": Object {
    "viewer": Object {
      "term": Object {
        "count": 200,
        "description": "",
        "name": "Watch This",
        "slug": "watch-this",
        "taxonomy": Object {
          "rewrite": Object {
            "slug": "category",
          },
        },
      },
    },
  },
}
`;

exports[`Test Term queries GraphQL should return a tag by slug and taxonomy 1`] = `
Object {
  "data": Object {
    "viewer": Object {
      "term": Object {
        "count": 3,
        "description": "",
        "name": "285 Kent",
        "slug": "285-kent",
        "taxonomy": Object {
          "rewrite": Object {
            "slug": "tag",
          },
        },
      },
    },
  },
}
`;
